Don't preinstall dependencies for now

deivid-rodriguez opened the pull request:
Bundler 1.16.0 is still giving us problems. Now we're hitting bundler/bundler#6154 and bundler/bundler#6162.

Basically, `bundler 1.16.0` generates binstubs with the path to the originating `Gemfile` hardcoded. That means that we end up with a bundler binstub at `/usr/local/bundle/bin/bundle` containing `/tmp/decidim_app/Gemfile` hardcoded, but that folder no longer exists in the image.

This seems to not have propagated yet to decidim's builds, but it's easily reproducible by logging into a built container of the image and running `bundle`. It leads to:

```
[!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - /tmp/decidim_app/Gemfile. Bundler cannot continue
```

You can also see a reproduction in my decidim fork (which had no base image caching, so hits the bug): https://circleci.com/gh/deivid-rodriguez/decidim/527.

Now, we have the following options:

* Remove preinstallation for now.
* Pin to an older image. Now I know a better way to do this now that building and pushing a local image, the one mention in [this comment](https://github.com/docker-library/ruby/issues/171#issuecomment-342710084):

```Dockerfile
FROM ruby@sha256:eed291437be80359321bf66a842d4d542a789e687b38c31bd1659065b2906778
...
```

* Manually downgrade bundler in our base `Dockerfile`. Workaround mentioned [here](https://github.com/bundler/bundler/issues/6162#issue-272343084):

```Dockerfile
RUN gem uninstall --all --executables --force --install-dir /usr/local/lib/ruby/gems/2.4.0 bundler && \
    gem install bundler --version 1.15.4

ENV BUNDLER_VERSION=1.15.4
```

Let me know what you think it's best!

mrcasals commented:
Considering that #5 was an improvement, but not a bug fix, I'm 👍 .

I'm not sure about the solution here. Looks like bundler 1.16 is very buggy, or at least we hit 3 bugs since its release, so what about downgrading it and keep the dependencies preinstall step?

Unfortunately @josepjaume is out of town these days, so I don't think he'll be able to chime in on this.

@deivid-rodriguez tell me what you think about downgrading bundler. If you think removing the dependencies preinstall step is the way to go I'll merge the PR 😄

deivid-rodriguez commented:
@mrcasals Yeah, bundler 1.16.0 turned out to be very buggy. This is normal with major bundler releases, but they usually release patch releases very quickly... Let's hope the project gets some traction again.

Anyways, as long as you manually explicitly set `BUNDLE_GEMFILE` (which I'm doing in decidim/decidim#2241), things work fine. So we could also leave things as they are for now. But yeah, probably downgrading `bundler` would be best.

mrcasals commented:
@deivid-rodriguez I'll leave it to you then 😄

deivid-rodriguez commented:
Let's leave this open for a bit a see what happens!

mrcasals commented:
@deivid-rodriguez just to confirm, "open" as in "not merged"?

deivid-rodriguez commented:
Yeah, open and unmerged!

deivid-rodriguez commented:
Unfortunately, this is not yet fixed. We're no longer corrupting the bundler binstub inside the image, but the one currently present is already corrupted (it contains `/tmp/decidim_app/Gemfile` hardcoded)...

I fixed this manually on my local image with `gem uninstall --all --executables --force --install-dir /usr/local/lib/ruby/gems/2.4.0 bundler` and then installing `bundler` again. But maybe it can be also fixed by fully rebuilding the image without caching, not sure.

josepjaume commented:
What I've seen people do is to include an `ENV BUILD_DATE=30/11/2017` at the beginning of an image so we can "trigger" a rebuild on a further date whenever we want. Should we do that?

josepjaume commented:
Check this out @deivid-rodriguez: https://github.com/decidim/docker/pull/11
